plugins/test/unit/plugins/distributors/yum/test_publish.py

```diff
@@ -501,7 +501,7 @@ def test_gpg_signing(self, _configuration, _RepomdXMLFileContext):
 
         self.assertEquals(_configuration.get_gpg_sign_options.return_value,
                           substep.sign_options)
-        # Make sure we're assing the sign options to the xml file context
```

Comment:
I meant "passing". I guess assigning would work too.

Reply:
OK, now I understand why test_utils.py didn't have to be updated.

In that test, self.sign_cmd is a string (not an array). as_environment() works fine in that case, because self.cmd is a string. However, that's not how you're invoking the code.

I think you need to keep self.cmd as a string (' '.join(cmd)) and assign self._cmdargs to cmd (which is now an array). That way, as_environment() will continue to put a meaningful string as GPG_CMD, as opposed to a strigified python list.

I am not sure how helpful it is for GPG_CMD to be present in the environment. When I initially coded this, I wanted the script to have its invocation string available as an environment variable. However the invocation string is available via argv or /proc/self/cmdline or similar ways. So, maybe, to keep things simple, we remove GPG_CMD as an environment variable?

I want to avoid going from array to string if it has the chance of creating bugs down the road, even if it's only informative in the environment.

I apologize for not catching this earlier.